### PR TITLE
FIXED: CPTabView was selecting the tab directly at mouse down whereas Cocoa does it at mouse up.

### PR DESCRIPTION
**Pull request by didierkorthoudt**
CPTabView was selecting the tab directly at mouse down whereas Cocoa does it at mouse up.

### CODE REVIEW

File: AppKit/CPTabView.j [hunk 1]
```diff
@@ -529,7 +531,7 @@ var CPTabViewDidSelectTabViewItemSelector           = 1 << 1,
 {
     var segmentIndex = [_tabs testSegment:[_tabs convertPoint:[anEvent locationInWindow] fromView:nil]];
 
-    if (segmentIndex != CPNotFound && [self selectTabViewItemAtIndex:segmentIndex])
```
[the remark this answers is not in the document]
Comment (didierkorthoudt) [reply AppKit/CPTabView.j]: Thank you for pointing this. I'll adapt ASAP.

[the remark this answers is not in the document]
Comment (didierkorthoudt) [reply AppKit/CPTabView.j]: Does the fix proposal is OK for you ? It works as expected on my tests.

**Comment by didierkorthoudt** [comment]
BTW, I've found another divergence from Cocoa : in modern Cocoa, when you select a tab (mouse down, holding) and drag upon another tab, the second one reacts and can then be selected by releasing the mouse over it. I'll propose a fix in another PR.

File: AppKit/CPTabView.j (hunk 1 above)
[the remark this answers is not in the document]
Comment (didierkorthoudt) [reply AppKit/CPTabView.j]: Sorry, forget this one. There's a divergence from Cocoa. In Cocoa, even if the delegate returns NO for souldSelect..., the tab highlights (not with this fix) but if you release the mouse, it just does nothing. I'll propose another fix.

[the remark this answers is not in the document]
Comment (cacaodev) [reply AppKit/CPTabView.j]: Actually, in cocoa it does not highlight but just shows a pressed state. This is the same issue you are mentioning bellow. I prefer that you leave this PR as it is and open a new one enabling the pressed state for CPSegmentedControl.

[the remark this answers is not in the document]
Comment (didierkorthoudt) [reply AppKit/CPTabView.j]: Indeed, it's a pushed state, not an highlight (selected item is highlighted).

OK then, I leave this one as such and go to the more general divergence.

The main problem is that we've built CPTabView with a CPSegmentedControl but in Cocoa, those two controls behave differently.

So this PR is now ready for review.

**Comment by didierkorthoudt** [comment]
Above divergence is fixed by PR #2694

**Comment by didierkorthoudt** [comment]
Closing this one as it will be included in PR #2694

**Pull request closed without merging**